src/Abc.Zebus/Directory/PeerDirectoryClient.cs

```diff
@@ -412,7 +367,18 @@ public void Handle(PeerSubscriptionsUpdated message)
 
             PeerUpdated?.Invoke(message.PeerDescriptor.PeerId, PeerUpdateAction.Updated);
 
-            DispatchSubscriptionsUpdatedMessages(message.PeerDescriptor.PeerId, message.PeerDescriptor.Subscriptions);
+            var observedSubscriptions = GetObservedSubscriptions(message.PeerDescriptor.Subscriptions);
+            if (observedSubscriptions.Length > 0)
+                PeerSubscriptionsUpdated?.Invoke(message.PeerDescriptor.PeerId, observedSubscriptions);
+        }
+
+        private Subscription[] GetObservedSubscriptions(Subscription[] peerDescriptorSubscriptions)
+        {
+            var observedSubscriptions = peerDescriptorSubscriptions?
+                                        .Where(x => _observedSubscriptionMessageTypes.Contains(x.MessageTypeId.GetMessageType()))
+                                        .ToArray()
```

Comment:
- `ToList` would be slightly better. We can still return `Array.Empty` when `peerDescriptorSubscriptions` is null.
- We should also directly return `Array.Empty` when `_observedSubscriptionMessageTypes` is empty (which will be true most of the time)